packages/core/src/shared/utilities/osUtils.ts

```diff
@@ -0,0 +1,61 @@
+/*!
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import globals from '../extensionGlobals'
+import { fs } from '../fs/fs'
+import { tempDirPath } from '../filesystemUtilities'
+import path from 'path'
+import { getExtensionId } from '../extensionUtilities'
+
+/**
+ * Checks if the current OS session is new.
+ *
+ * @returns `true` if this is the first call to this function across all extension instances
+ * since the OS was last restarted, `false` otherwise.
+ *
+ * Use this function to perform one-time initialization tasks that should only happen
+ * once per OS session, regardless of how many extension instances are running.
+ */
+export async function isNewOsSession(tmpDir = tempDirPath) {
+    switch (process.platform) {
+        case 'win32': {
+            // Windows does not have an ephemeral /tmp/ folder that deletes on shutdown, while unix-like os's do.
+            // So in Windows we calculate the start time and see if it changed from the previous known start time.
+
+            const lastStartTime = globals.globalState.get<number>('lastOsStartTime', undefined)
```

Comment:
tryGet gives some extra type-checking protection (this is mentioned in the get() docstring)
```suggestion
            const lastStartTime = globals.globalState.tryGet('lastOsStartTime', Number)
```

Reply:
ty!